hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/TestInProcessLockProvider.java

```diff
@@ -160,6 +160,48 @@ public void testTryLockReAcquisitionByDifferentThread() {
     });
   }
 
+  @Test
+  public void testTryUnLockByDifferentThread() {
+    InProcessLockProvider inProcessLockProvider = new InProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    final AtomicBoolean writer2Completed = new AtomicBoolean(false);
+    final AtomicBoolean writer3Completed = new AtomicBoolean(false);
+
+    // Main test thread
+    Assertions.assertTrue(inProcessLockProvider.tryLock());
+
+    // Another writer thread
+    Thread writer2 = new Thread(() -> {
```

Comment:
why need writer2 when you have writer3 below?

Reply:
sorry, some refactoring residues. will clean it up.

Reply:
here is the context 
writer2 unlocks w/o actually locking. if a non owner unlocks, we don't throw any exception as such. but intrinsically, the lock is still acquired by the original owner (wirter1)

and so, we again try to acquire the lock by writer3. if at all the unlocking by writer2 succeeded, writer3 should be able to acquire the lock. but we expect it to fail since writer1 is the orignal owner of the lock. 

and hence we need 3 writers to verify this scenario.